dashboard/src/main/resources/templates/macros.ftl

```diff
@@ -8,23 +8,25 @@
   <#return plural?string(pluralNoun, singularNoun)>
 </#function>
 
-<#macro formatJarLinkageReport jar problemsToClassFiles jarToDependencyPaths dependencyPathRootCauses>
-  <!-- problemsToClassFiles: ImmutableMap<SymbolProblem, Set<String>> -->
-  <#if problemsToClassFiles?size gt 0>
+<#macro formatJarLinkageReport jar problemsWithClass jarToDependencyPaths dependencyPathRootCauses>
+  <!-- problemsToClasses: ImmutableSetMultimap<SymbolProblem, String> converted to
+    ImmutableMap<SymbolProblem, Collection<String>>; Multimaps don't work well with Freemarker -->
+  <#assign problemsToClasses = problemsWithClass.asMap() />
+  <#if problemsToClasses?size gt 0>
     <h3>${jar.getFileName()?html}</h3>
-    <#assign symbolProblemCount = problemsToClassFiles?keys?size />
+    <#assign symbolProblemCount = problemsToClasses?keys?size />
     <#assign sourceClassCount = 0 />
-    <#list problemsToClassFiles?values as classFiles>
-      <#assign sourceClassCount = sourceClassCount + classFiles?size />
+    <#list problemsToClasses as symbolProblem, sourceClasses>
+      <#assign sourceClassCount = sourceClassCount + sourceClasses?size />
```

Comment:
This won't be the number of unique source classes, since if a source class has more than one problem it'll appear more than once.

Maybe you want `problemsWithClass.inverse().keySet().size()`?